### ISSUE TITLE
[ML-v0.2] Intégration du modèle LSTM pour prédiction des prix futurs

### ISSUE THREAD

**Issue by Y0nyx** [Y0nyx/Agentic-Trader#15]
### 🎯 Objectif
Intégrer un modèle prédictif LSTM (Long Short-Term Memory) robuste pour prévoir les prix futurs des actifs financiers et améliorer significativement les performances des stratégies de trading existantes.

#### 1. **Préparation des données (data_preprocessing.py)**
- [ ] Fonction `prepare_lstm_data()` qui :
  - Charge les données CSV depuis `data/csv/` (colonnes : date, open, high, low, close, volume)
  - Crée des séquences temporelles (fenêtrage) avec `sequence_length=60` par défaut
  - Normalise les données avec `MinMaxScaler` de scikit-learn (valeurs entre 0 et 1)
  - Split train/validation/test : 70%/15%/15%
  - Retourne : `(X_train, y_train, X_val, y_val, X_test, y_test, scaler)`

#### 2. **Architecture du modèle LSTM (lstm_model.py)**
- [ ] Classe `LSTMPricePredictor` avec architecture suivante :
  ```python
  # Architecture suggérée
  - Input layer: (sequence_length, n_features)
  - LSTM layer 1: 128 units, return_sequences=True, dropout=0.2
  - LSTM layer 2: 64 units, dropout=0.2
  - Dense layer 1: 32 units, activation='relu'
  - Output layer: 1 unit (prix prédit)
  - Optimizer: Adam(learning_rate=0.001)
  - Loss: MSE (Mean Squared Error)
  ```

#### 3. **Entraînement et sauvegarde (train_lstm.py)**
- [ ] Fonction `train_model()` qui :
  - Entraîne le modèle avec early stopping (patience=10)
  - Sauvegarde automatique du meilleur modèle dans `models/lstm_best.h5`
  - Log les métriques d'entraînement dans `logs/lstm_training.log`
  - Génère un graphique de l'historique d'entraînement (loss curves)

#### 4. **Évaluation des performances (evaluate_lstm.py)**
- [ ] Fonction `evaluate_predictions()` calculant :
  - RMSE (Root Mean Square Error)
  - MAE (Mean Absolute Error)
  - Précision directionnelle (% de prédictions correctes de hausse/baisse)
  - Graphique comparatif : prix réels vs prédictions

#### 5. **Intégration avec les stratégies existantes (lstm_strategy.py)**
- [ ] Classe `LSTMEnhancedStrategy` qui :
  - Hérite de la stratégie de base
  - Utilise les prédictions LSTM pour ajuster les signaux BUY/SELL/HOLD
  - Paramètre de confiance configurable (threshold pour utiliser la prédiction)

### 📁 Structure de fichiers attendue
```
agentic_trader/
├── ml_models/
│   ├── __init__.py
│   ├── data_preprocessing.py
│   ├── lstm_model.py
│   ├── train_lstm.py
│   ├── evaluate_lstm.py
│   └── lstm_strategy.py
├── models/
│   └── lstm_best.h5 (généré après entraînement)
├── logs/
│   └── lstm_training.log (généré)
└── tests/
    ├── test_data_preprocessing.py
    ├── test_lstm_model.py
    └── test_lstm_strategy.py
```

#### Tests à implémenter :
```python
# test_data_preprocessing.py
def test_sequence_creation():
    """Vérifie la création correcte des séquences temporelles"""
    
def test_data_normalization():
    """Vérifie que les données sont bien normalisées entre 0 et 1"""
    
def test_train_test_split():
    """Vérifie les proportions du split 70/15/15"""

# test_lstm_model.py
def test_model_architecture():
    """Vérifie que l'architecture du modèle est correcte"""
    
def test_model_compilation():
    """Vérifie la compilation avec les bons paramètres"""

# test_lstm_strategy.py
def test_signal_generation_with_predictions():
    """Vérifie l'intégration des prédictions dans les signaux"""
```

**Couverture minimale attendue : 90%**

### 🧪 Exemple d'utilisation complet
```python
# Exemple d'utilisation du module LSTM
from ml_models.data_preprocessing import prepare_lstm_data
from ml_models.lstm_model import LSTMPricePredictor
from ml_models.train_lstm import train_model
from ml_models.evaluate_lstm import evaluate_predictions
from ml_models.lstm_strategy import LSTMEnhancedStrategy

# 1. Préparation des données
X_train, y_train, X_val, y_val, X_test, y_test, scaler = prepare_lstm_data(
    filepath="data/csv/BTC-USD.csv",
    sequence_length=60
)

# 2. Création et entraînement du modèle
model = LSTMPricePredictor(sequence_length=60, n_features=5)
history = train_model(model, X_train, y_train, X_val, y_val, epochs=100)

# 3. Évaluation
metrics = evaluate_predictions(model, X_test, y_test, scaler)
print(f"RMSE: {metrics['rmse']:.4f}")
print(f"Précision directionnelle: {metrics['directional_accuracy']:.2%}")

# 4. Utilisation dans une stratégie
strategy = LSTMEnhancedStrategy(
    model_path="models/lstm_best.h5",
    confidence_threshold=0.7
)
signals = strategy.generate_signals(df)
```

### 🧰 Dépendances spécifiques
```txt
tensorflow==2.15.0
scikit-learn==1.3.2
pandas==2.1.4
numpy==1.24.3
matplotlib==3.8.2
```

### 📊 Métriques de performance attendues
- RMSE < 5% du prix moyen de l'actif
- MAE < 3% du prix moyen de l'actif
- Précision directionnelle > 55%
- Temps d'entraînement < 30 minutes sur GPU standard

### 🎯 Définition de terminé (DoD)
- [ ] Code implémenté selon les spécifications ci-dessus
- [ ] Tests unitaires passants avec couverture ≥ 90%
- [ ] Documentation complète (docstrings pour toutes les fonctions publiques)
- [ ] Exemple fonctionnel dans le README
- [ ] Revue de code approuvée
- [ ] Performances validées sur au moins 3 actifs différents (BTC, AAPL, EUR/USD)

---

**Priorité :** 🔴 Haute  
**Version :** v0.2  
**Assigné à :** @Y0nyx  
**Labels :** ML, high-priority, testing, v0.2, ai-ready
